gcloud-java-core/src/main/java/com/google/cloud/HttpServiceOptions.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.cloud.spi.ServiceRpcFactory;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.Objects;
+
+/**
+ * Abstract class representing service options for those services that use HTTP as the transport
+ * layer.
+ *
+ * @param <ServiceT> the service subclass
+ * @param <ServiceRpcT> the spi-layer class corresponding to the service
+ * @param <OptionsT> the {@code ServiceOptions} subclass corresponding to the service
+ */
+public abstract class HttpServiceOptions<ServiceT extends Service<OptionsT>, ServiceRpcT,
+    OptionsT extends HttpServiceOptions<ServiceT, ServiceRpcT, OptionsT>>
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Generics names should be a one letter long and upper case.](https://www.codacy.com/app/mziccard/gcloud-java/file/2575979929/issues/source?bid=3123109&fileBranchId=3299216#l42)